compiler/infra/deque.hpp

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
```

Comment:
OK.  I figured it was something like that.  As there isn't a documented rule for file names (only a loose convention) I think this is fine.